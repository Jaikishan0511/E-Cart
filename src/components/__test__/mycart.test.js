import MyCart from "../index";
import React from "react";
import ReactTestRenderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
const MyCartComp = (
    <Router>
      <MyCart />
    </Router>
  );
describe("Actions", () => {
    describe("Render", () => {
      it("should be a valid React Element", () => {
        expect(React.isValidElement(<MyCart />)).toBeTruthy();
      });
  
      it("should render without throwing an error", () => {
        expect(() =>
          ReactTestRenderer.create(MyCartComp)
        ).not.toThrow();
      });
    });
  
    describe("Snapshots", () => {
      it("matches the default snapshot", () => {
        const tree = ReactTestRenderer.create(MyCartComp).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });