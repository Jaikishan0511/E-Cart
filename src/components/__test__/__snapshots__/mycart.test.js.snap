// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions Snapshots matches the default snapshot 1`] = `
<div>
  <div
    className="HomePage"
  >
    <div
      className="header-container"
    >
      <div
        className="title"
      >
         MyCart 
      </div>
      <div
        className="search-bar"
      >
        <img
          alt="logo"
          src="/images/search-icon.png"
        />
        <input
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="cart-img"
      >
        <a
          href="/view-cart"
          onClick={[Function]}
        >
          <img
            alt="logo"
            src="/images/cart.png"
          />
          <span
            className="badge"
          >
            0
          </span>
        </a>
      </div>
      <div />
    </div>
    <div
      className="page-content"
    >
      <div
        className="multiple-categories"
      >
        <div
          className="close-category-icon"
          onClick={[Function]}
        >
          <img
            alt="close-icon"
            src="/images/close-icon.png"
          />
        </div>
        <div
          className="categories-label"
        >
          <input
            defaultChecked={true}
            id={0}
            name="formHorizontalRadios"
            onClick={[Function]}
            type="radio"
            value="All"
          />
          <label
            className="radio-lable"
          >
            All
          </label>
        </div>
        <div
          className="categories-label"
        >
          <input
            defaultChecked={false}
            id={1}
            name="formHorizontalRadios"
            onClick={[Function]}
            type="radio"
            value="Electronics"
          />
          <label
            className="radio-lable"
          >
            Electronics
          </label>
        </div>
        <div
          className="categories-label"
        >
          <input
            defaultChecked={false}
            id={2}
            name="formHorizontalRadios"
            onClick={[Function]}
            type="radio"
            value="Food Essentials"
          />
          <label
            className="radio-lable"
          >
            Food Essentials
          </label>
        </div>
        <div
          className="categories-label"
        >
          <input
            defaultChecked={false}
            id={3}
            name="formHorizontalRadios"
            onClick={[Function]}
            type="radio"
            value="Home & furniture"
          />
          <label
            className="radio-lable"
          >
            Home & furniture
          </label>
        </div>
        <div
          className="categories-label"
        >
          <input
            defaultChecked={false}
            id={4}
            name="formHorizontalRadios"
            onClick={[Function]}
            type="radio"
            value="Fashion"
          />
          <label
            className="radio-lable"
          >
            Fashion
          </label>
        </div>
        <div
          className="categories-label"
        >
          <input
            defaultChecked={false}
            id={5}
            name="formHorizontalRadios"
            onClick={[Function]}
            type="radio"
            value="Sports"
          />
          <label
            className="radio-lable"
          >
            Sports
          </label>
        </div>
      </div>
      <div
        className="product-filter"
      >
        <div
          className="Products-list"
        />
      </div>
    </div>
  </div>
</div>
`;
